### PR TITLE
validate that a servers ip6address field is an IPv6 address or CIDR

### LINKED ISSUE #2790: cannot add / for v6 ip in servers field. Need to sort alphabetically on location
cannot add a / to v6 ip's on server add page. error is / is not allowed.
choosing site  on server add page is a random order, with over 50 sites, it's painful to navigate.

### PR DESCRIPTION
**Pull request by DylanVolz**
#### What does this PR do?

Fixes #2790

#### Which TC components are affected by this PR?

- [ ] Documentation
- [ ] Grove
- [ ] Traffic Analytics
- [ ] Traffic Monitor
- [X] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] Other _________

#### What is the best way to verify this PR?

create or update a server with an ipv6 CIDR in the ipv6address field. (the api tests do this)

#### Check all that apply

- [X] This PR includes tests
- [ ] This PR includes documentation updates
- [ ] This PR includes an update to CHANGELOG.md
- [ ] This PR includes all required license headers
- [ ] This PR includes a database migration (ensure that migration sequence is correct)
- [ ] This PR fixes a serious security flaw. Read more: [www.apache.org/security](http://www.apache.org/security/)

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2388/
Test PASSed.

File: lib/go-tc/tovalidate/rules.go [hunk 1]
```diff
@@ -129,3 +130,33 @@ func IsValidPortNumber(value interface{}) error {
...
+		ip, _, err := net.ParseCIDR(*v)
+		if err == nil {
+			if ip.To4() == nil {
+				return nil
+			} else {
+				return fmt.Errorf("got IPv4 CIDR, IPv6 expected")
+			}
+		} else {
+			ip := net.ParseIP(*v)
+			if ip != nil {
```
Comment (dewrich) [inline lib/go-tc/tovalidate/rules.go]: You can't combine this nested if?

File: lib/go-tc/tovalidate/rules.go [hunk 2]
```diff
@@ -129,3 +130,33 @@ func IsValidPortNumber(value interface{}) error {
 	}
 	return errors.New("must be a valid port number")
 }
+
+func IsValidIPv6CIDROrAddress(value interface{}) error {
```
Comment (dewrich) [inline lib/go-tc/tovalidate/rules.go]: What about IPv4 in this func name? `IsValidIPv6CIDROrAddress`

File: lib/go-tc/tovalidate/rules.go (hunk 1 above)
Comment (DylanVolz) [reply lib/go-tc/tovalidate/rules.go]: Yeah, we can because we don't care about the difference between an un-parseable string and an IPv4 address.

**Pull request merged**